build.gradle

```diff
@@ -535,9 +543,9 @@ task coverage {
 
 task release {
     if (isLegacyBuild) {
-        dependsOn build, test, javadoc, src_jar, checkstyle
```

Comment:
should we invoke spotless here instead?

Reply:
No, Spotless registers itself as a dependency on the `check` step which is a dependency of `build`. The reason `checkstyle` was declared here was probably because we implemented the task by hand and didn't completely follow Gradle convention.

Reply:
hm, it would be nice to only invoke spotless on `release` so that we can iterate quickly (if un-stylishly) with `./gradlew cmake_clean build`. but i guess it's not too bad to just apply the fixes inline with `./gradlew cmake_clean spotlessApply build` and then refresh your editor pane.

> Spotless registers itself as a dependency on the check step which is a dependency of build

what does `check` do?

Reply:
I think having spotless automate formatting is painless enough to allow us to make it a required part of the build. It's not as annoying as checkstyle or findbugs which would just break your build and force you to either look up how to bypass them or figure out how to fix your code.

Here's a blurb from the Gradle docs about `check`.

>   check — lifecycle task -  Plugins and build authors should attach their verification tasks, such as ones that run tests, to this lifecycle task using check.dependsOn(task).

https://docs.gradle.org/current/userguide/base_plugin.html#sec:base_tasks

Reply:
ok, cool. TIL about `-x`, very useful.